aemedge/blocks/accordion/accordion.js

```diff
@@ -12,22 +15,26 @@ export default function decorate(block) {
   [...block.children].forEach((row) => {
     // decorate accordion item label
     const label = row.children[0];
-    const summary = document.createElement('summary');
-    summary.className = 'accordion-item-label';
-    summary.append(...label.childNodes);
-    if (!hasWrapper(summary)) {
-      summary.innerHTML = `<p>${summary.innerHTML}</p>`;
-    }
+    const summaryArrow = span({ class: 'icon icon-slim-arrow-right accordion-arrow' });
+    const summaryContents = [summaryArrow, ...label.childNodes];
+    const summary = domEl(
+      'summary',
+      { class: 'accordion-item-label' },
+      ...(!hasWrapper(label) ? [p(...summaryContents)] : summaryContents),
+    );
+
     // decorate accordion item body
-    const body = row.children[1];
+    const body = !hasWrapper(row.children[1]) ? p(...row.children[1]) : row.children[1];
     body.className = 'accordion-item-body';
-    if (!hasWrapper(body)) {
-      body.innerHTML = `<p>${body.innerHTML}</p>`;
-    }
+
     // decorate accordion item
-    const details = document.createElement('details');
-    details.className = 'accordion-item';
-    details.append(summary, body);
+    const details = domEl(
+      'details',
+      { class: 'accordion-item' },
+      summary,
+      body,
+    );
+    decorateIcons(details);
```

Comment:
Can we run it outside the loop? May be call `decorateIcons(block)`? It should be able to decorate all icons inside

Reply:
+1